api/client/commands.go

```diff
@@ -2263,6 +2264,21 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 	if err != nil {
 		utils.ReportError(cmd, err.Error(), true)
 	}
+
+	if len(hostConfig.Dns) > 0 {
+		// check the DNS settings passed via --dns against
+		// localhost regexp to warn if they are trying to
+		// set a DNS to a localhost address
+		hasLocalhost := false
+		for _, dnsIP := range hostConfig.Dns {
+			if resolvconf.IsLocalhost(dnsIP) {
+				hasLocalhost = true
```

Comment:
@estesp can you move the print statement here, and do a break, it should remove the need for the variable altogether.